packages/ra-core/src/reducer/admin/references/oneToMany.ts

```diff
@@ -114,6 +138,45 @@ export const getReferencesByIds = (
     return Object.keys(references).length > 0 ? references : null;
 };
 
+const getRelatedReferences = (previousState, resource) =>
+    Object.keys(previousState).filter(key => key.includes(resource));
+
+const removeDeletedReference = (removedId: Identifier) => (
+    previousState,
+    key
+) => {
+    const hasReferenceToRemovedId = previousState[key].ids.includes(removedId);
+
+    if (!hasReferenceToRemovedId) {
+        return previousState;
+    }
+
+    return {
+        ...previousState,
+        [key]: {
+            ids: previousState[key].ids.filter(id => id !== removedId),
+            total: previousState[key].total - 1,
+        },
+    };
+};
+
+const removeDeletedReferences = (removedIds: Identifier[]) => (
+    previousState,
+    key
+) => {
+    const idsToKeep = previousState[key].ids.filter(
+        id => !removedIds.includes(id)
+    );
+
```

Comment:
again, you need to check that there is something to do before creating another state

Reply:
You mean before recreating a state equal to the previous one ?